hr_maintenance_workspaces/models/equipment.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 SDi Digital Group (https://www.sdi.es/odoo-cloud)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class MaintenanceEquipment(models.Model):
+    _inherit = 'maintenance.equipment'
+    _sql_constraints = [
+        (
+            'internal_reference_unique_equipment',
+            'unique(internal_reference)',
+            'Internal Reference already exists!',
+        ),
+    ]
+
+    internal_reference = fields.Char(
+        string='Internal Reference',
+        copy=False,
+    )
+    workspace_id = fields.Many2one(
+        comodel_name='maintenance.workspace',
+        string='Workspace',
+        tracking=True,
+    )
+    equipment_assign_to = fields.Selection(
+        selection_add=[('workspace', 'Workspace'), ('other', 'Other')],
+        ondelete={'workspace': 'set default'},
+    )
+
+    @api.depends('equipment_assign_to')
+    def _compute_equipment_assign(self):
+        for equipment in self:
+            if equipment.equipment_assign_to == 'employee':
+                equipment.department_id = False
+                equipment.workspace_id = False
+                equipment.employee_id = equipment.employee_id
+            elif equipment.equipment_assign_to == 'department':
+                equipment.employee_id = False
+                equipment.workspace_id = False
+                equipment.department_id = equipment.department_id
+            elif equipment.equipment_assign_to == 'workspace':
+                equipment.workspace_id = equipment.workspace_id
+                equipment.department_id = False
+                equipment.employee_id = False
+            else:
+                equipment.workspace_id = equipment.workspace_id
+                equipment.department_id = equipment.department_id
+                equipment.employee_id = equipment.employee_id
+            equipment.assign_date = fields.Date.context_today(self)
```

Comment:
You don't need to re-implement the method here. You can call `super()._compute_equipment_assign()` to run the parent method and when it returns handle workspace related modifications.